### PR TITLE
Tickbase and Resolver

### PR DESCRIPTION
**Pull request by ohkriss**
Add initial resolver to this branch and fix a compiler error caused in Tickbase.h

### CODE REVIEW

**Comment by smokez129** [comment]
i get errors Cannot open include file: 'Hacks/Resolver.h': No such file or directory and Severity	Code	cannot open source file "Hacks/Resolver.h"	

i changed all code that had to be changed

**Comment by 2933968621** [comment]
?

**Comment by r3muxd** [comment]
itd probably be better to cycle through -60, 60, 30, and -30
and then use antifreestanding to detect side (or animlayers)

**Comment by r3muxd** [comment]
(and clamp to max desync angle)

**Comment by ohkriss** [comment]
> itd probably be better to cycle through -60, 60, 30, and -30 and then use antifreestanding to detect side (or animlayers)

I will test this

**Comment by smokez129** [comment]
i get error  cannot open source file "Hacks/Resolver.h"

and when i paste              if (stage == FrameStage::NET_UPDATE_POSTDATAUPDATE_START) {
            Resolver::frameStageUpdate();
        }

i get error  name followed by '::' must be a class or namespace name

**Comment by smokez129** [comment]
i get error cannot open source file "Hacks/Resolver.h"

and when i paste  if (stage == FrameStage::NET_UPDATE_POSTDATAUPDATE_START) {
            Resolver::frameStageUpdate();
        }

i get error    name followed by '::' must be a class or namespace name

**Comment by ohkriss** [comment]
> i get error cannot open source file "Hacks/Resolver.h"
> 
> and when i paste if (stage == FrameStage::NET_UPDATE_POSTDATAUPDATE_START) { Resolver::frameStageUpdate(); }
> 
> i get error name followed by '::' must be a class or namespace name

This is cause by some error on your end, cause I wasn't able to reproduce this issue. If someone else is getting this error, let me know

**Comment by ohkriss** [comment]
@notgoodusename Can you make a new branch for developing the resolver where I can get write access? Or should I move this pr to the main branch, since this branch was intended for tickbase development?

**Comment by Yoka1998** [comment]
Bro help me i don't know how to open "Osiris.vcxproj" , "Osiris.vcxproj.filters" and "Osiris.vcxproj.user"

File: Osiris/Hacks/Resolver.cpp [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+
+        entity->updateState(animstate, entity->eyeAngles());
+
+        int missed = record->missedshots;
+        if (record->lastworkingshot != -1)
+            missed = record->lastworkingshot;
+
+        switch (missed % 9) {
+        case 1:
+            desyncAng += 30.0f;
```
Comment (voidzer0-development) [inline Osiris/Hacks/Resolver.cpp:133]: I'd use a fraction of max desync angle instead of static 30

File: Osiris/Hacks/Resolver.cpp [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+            break;
+        case 2:
+            desyncAng -= 30.0f;
+            break;
+        case 3:
+            desyncAng += 60.0f;
```
Comment (voidzer0-development) [inline Osiris/Hacks/Resolver.cpp:139]: If the player velocity is too high it is impossible to desync 60 deg

File: Osiris/Hacks/Resolver.cpp [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+
+        auto animstate = entity->getAnimstate();
+
+        if (!animstate)
+            return;
+
+        if (!record->FiredUpon || !record->wasTargeted) {
+
+            entity->updateState(animstate, entity->eyeAngles());
+            if ((record->wasUpdated == false) && (entity->eyeAngles().y != record->PreviousEyeAngle)) {
```
Comment (voidzer0-development) [inline Osiris/Hacks/Resolver.cpp:103]: The real yaw can change without the fake yaw changing

File: Osiris/Hacks/Resolver.cpp [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+            return;
+        }
+
+        entity->updateState(animstate, entity->eyeAngles());
+
+        int missed = record->missedshots;
+        if (record->lastworkingshot != -1)
+            missed = record->lastworkingshot;
+
+        switch (missed % 9) {
```
Comment (voidzer0-development) [inline Osiris/Hacks/Resolver.cpp:131]: Where side detection? Even something basic like the following would be better
`float difference = AngleDiff ( eyeYaw, record->animstate->m_flFootYaw );
            desyncSide = 2 * difference <= 0.0f ? 1 : -1;`

**Comment by Africanized** [comment]
game crashes when picking a team

**Comment by cristyanul** [comment]
How is the resolver connected to ragebot/ bone matrixes?

**Pull request closed without merging**